versions-common/src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java

```diff
@@ -479,13 +464,21 @@ private List<IgnoreVersion> getIgnoredVersions( Artifact artifact )
 
     @Override
     public void resolveArtifact( Artifact artifact, boolean usePluginRepositories )
-        throws ArtifactResolutionException, ArtifactNotFoundException
+        throws ArtifactResolutionException
     {
-        artifactResolver.resolve( artifact,
-                usePluginRepositories
-                        ? mavenSession.getCurrentProject().getPluginArtifactRepositories()
-                        : mavenSession.getCurrentProject().getRemoteArtifactRepositories(),
-                localRepository );
+        try
+        {
+            aetherRepositorySystem.resolveArtifact( mavenSession.getRepositorySession(),
+                    new ArtifactRequest( toArtifact( artifact ),
+                            usePluginRepositories
+                                    ? mavenSession.getCurrentProject().getRemotePluginRepositories()
+                                    : mavenSession.getCurrentProject().getRemoteProjectRepositories(),
+                            getClass().getName() ) );
```

Comment:
We need process response and update source artifact, like:
```
rtifactResult artifactResult  = resolveArtifact( ... );

        artifact.setFile( artifactResult.getArtifact().getFile() );
        artifact.setVersion( artifactResult.getArtifact().getVersion() );
        artifact.setResolved( true );
```

Reply:
Done